common/dynamicconfig/constants.go

```
@@ -713,6 +713,11 @@ This config is EXPERIMENTAL and may be changed or removed in a later release.`,
 		10,
 		`FrontendMaxBadBinaries is the max number of bad binaries in namespace config`,
 	)
+	FrontendHideInternalOrUnknownErrors = NewGlobalBoolSetting(
+		"frontend.hideInternalErrors",
```

I would rename it to `frontend.hideErrorDetails`. Who knows what codes will be there tomorrow.

---

for me "hide error details" imply that _all_ error details are hidden. We are not hiding all errors, just a few specific

---

"Internal" conflicts with `serviceerror.Internal`. I would think that this parameter is for them only. Not for any other status code.

---

not sure I understand. That parameter is for errors with Internal or Unknown status codes